src/App.tsx

```diff
@@ -15,58 +15,101 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type State = {
+  selectedGood: string;
+};
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
+export class App extends React.Component<{}, State> {
+  state = {
+    selectedGood: 'Jam',
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+  clearSelectedGood = () => {
+    this.setState({
+      selectedGood: '',
+    });
+  };
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  addSelectedGood = (newSelectedGood: string) => {
+    this.setState({
+      selectedGood: newSelectedGood,
+    });
+  };
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  mapGoods = (selected: string) => {
+    const { selectedGood } = this.state;
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+    const addButton = (
+      <button
+        data-cy="AddButton"
+        type="button"
+        className="button"
+        onClick={() => {
+          this.addSelectedGood(selected);
+        }}
+      >
+        +
+      </button>
+    );
 
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+    const removeButton = (
+      <button
+        data-cy="RemoveButton"
+        type="button"
+        className="button is-info"
+        onClick={this.clearSelectedGood}
+      >
+        -
+      </button>
+    );
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Garlic
-          </td>
-        </tr>
-      </tbody>
-    </table>
-  </main>
-);
+    return (
+      <tr
+        data-cy="Good"
+        className={
+          selectedGood === selected ? 'has-background-success-light' : ''
+        }
+        key={selected}
+      >
+        <td>{selectedGood !== selected ? addButton : removeButton}</td>
```

Comment:
A bit of overkill here, you can have just one button and change it's attributes:

```
      <button
        data-cy="... ? .... : ...."
        type="button"
        className="button ... ? .... : ...."
        onClick={... ? .... : ....}
      >
       {... ? .... : ....}
      </button>
```